File: app/src/main/java/com/tangem/wallet/NoExtendedLengthSupportDialog.java
package com.tangem.wallet;

import android.app.AlertDialog;
import android.app.Dialog;
import android.app.DialogFragment;
import android.content.DialogInterface;
import android.os.Bundle;

public class NoExtendedLengthSupportDialog extends DialogFragment {

    public static boolean allreadyShowed=false;

    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {

        return new AlertDialog.Builder(getActivity())
                .setIcon(R.drawable.tangem_logo_small_new)
                .setTitle("Warning")
                .setMessage("The NFC adapter of the device does not support extended length APDU, it's possible that some functions will not work!")
                .setPositiveButton("Got it",
                        new DialogInterface.OnClickListener() {
                            public void onClick(DialogInterface dialog, int whichButton) {
                                NoExtendedLengthSupportDialog.allreadyShowed=true;
                            }
                        }
                )
                .create();
    }

    @Override
    public void onCancel(DialogInterface dialog) {
        super.onCancel(dialog);
    }
}

File: app/src/main/java/com/tangem/ui/dialog/NoExtendedLengthSupportDialog.kt
package com.tangem.ui.dialog

import android.app.AlertDialog
import android.app.Dialog
import android.os.Bundle
import androidx.fragment.app.DialogFragment

import com.tangem.wallet.R

class NoExtendedLengthSupportDialog : DialogFragment() {

    companion object {
        val TAG: String = NoExtendedLengthSupportDialog::class.java.simpleName
        var allReadyShowed = false
        var message = ""// = R.string.the_nfc_adapter_length_apdu
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        return AlertDialog.Builder(activity)
                .setIcon(R.drawable.tangem_logo_small_new)
                .setTitle(R.string.dialog_warning)
                .setMessage(message)
                .setPositiveButton(R.string.dialog_btn_got_it) { _, _ -> NoExtendedLengthSupportDialog.allReadyShowed = false }
                .create()
    }

}